src/main/java/com/google/sps/model/AuthenticatedHttpServlet.java

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.model;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.sps.utility.AuthenticationUtility;
+import java.io.IOException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** HttpServlet that enforces the verification of user tokens */
+public abstract class AuthenticatedHttpServlet extends HttpServlet {
+  // Error message if user is not authenticated
+  protected static final String ERROR_403 = "Authentication tokens not present / invalid";
+
+  protected Credential googleCredential;
```

Comment:
Instead of storing googleCredential as an instance variable, I suggest creating a new abstract `doGet` that takes a Credential as a parameter. You would call it only if there are valid credentials.

This has two benefits:

1. You won't forget to check if `googleCredential` is valid in your servelet.
2. It's really easy to test a servelet by just calling its `doGet` with a mock credential.

Reply:
Done! Really cool - thanks